onadata/apps/api/viewsets/connect_viewset.py

```diff
@@ -42,8 +44,11 @@ def user_profile_w_token_response(request, status):
     try:
         user_profile = request.user.profile
     except UserProfile.DoesNotExist:
-        user_profile, _ = UserProfile.objects.get_or_create(
-            user=request.user)
+        user_profile = cache.get(
+            f'{USER_PROFILE_PREFIX}{request.user.username}')
+        if not user_profile:
+            user_profile, __ = UserProfile.objects.get_or_create(
+                user=request.user)
```

Comment:
Should we cache the profile here as well after retrieving it? so that we don't have the db again if a new requests is made before the cache expires.

Reply:
This just means that `user_profile` won't be empty and we won't have to hit the db

Reply:
Made this change.